R/shell_connection.R

```diff
@@ -546,34 +547,82 @@ initialize_connection.spark_shell_connection <- function(sc) {
       }
 
       context_config <- connection_config(sc, "spark.", c("spark.sql."))
-      apply_config(context_config, conf, "set", "spark.")
+      apply_config(conf, context_config, "set", "spark.")
 
       default_config <- shell_connection_config_defaults()
       default_config_remove <- Filter(function(e) e %in% names(context_config), names(default_config))
       default_config[default_config_remove] <- NULL
-      apply_config(default_config, conf, "set", "spark.")
+      apply_config(conf, default_config, "set", "spark.")
 
       # create the spark context and assign the connection to it
-      sc$spark_context <- invoke_static(
-        sc,
-        "org.apache.spark.SparkContext",
-        "getOrCreate",
-        conf
-      )
 
-      invoke(backend, "setSparkContext", sc$spark_context)
-    }
+      sc$state$spark_context <- if (spark_version(sc) >= "2.0") {
+
```

Comment:
We need to create a `SparkSession` anyway so it doesn't hurt performance to cache.